caas/kubernetes/provider/k8s_test.go

```diff
@@ -276,19 +313,42 @@ func (s *K8sBrokerSuite) TestDestroy(c *gc.C) {
 	ctrl := s.setupBroker(c)
 	defer ctrl.Finish()
 
-	// Delete operations below return a not found to ensure it's treated as a no-op.
+	ns := &core.Namespace{ObjectMeta: v1.ObjectMeta{Name: "test"}}
+	namespaceWatcher := s.k8sNewFakeWatcher()
+
 	gomock.InOrder(
+		s.mockNamespaces.EXPECT().Watch(
+			v1.ListOptions{
+				FieldSelector:        fields.OneTermEqualSelector("metadata.name", "test").String(),
+				IncludeUninitialized: true,
+			},
+		).
+			Return(namespaceWatcher, nil),
 		s.mockNamespaces.EXPECT().Delete("test", s.deleteOptions(v1.DeletePropagationForeground)).Times(1).
-			Return(s.k8sNotFoundError()),
+			Return(nil),
 		s.mockStorageClass.EXPECT().DeleteCollection(
 			s.deleteOptions(v1.DeletePropagationForeground),
 			v1.ListOptions{LabelSelector: "juju-model==test"},
 		).Times(1).
 			Return(s.k8sNotFoundError()),
+		s.mockNamespaces.EXPECT().Get("test", v1.GetOptions{IncludeUninitialized: true}).Times(1).
```

Comment:
This doesn't look right? We expect either nil if namespace still exists, or not found when it goes away. Any other error should cause Destroy() to fail

Reply:
yes, fixed!